iiblib/iib_build_details_model.py

```diff
@@ -1,135 +1,69 @@
-            data["index_image"],
-            data["request_type"],
-            data["arches"],
-            data["bundle_mapping"],
-            data.get("omps_operator_version", {}),
-        )
+        if data['request_type'] == "add":
+            return AddModel._from_dict(data)
+        elif data['request_type'] == "rm":
+            return RmModel._from_dict(data)
+        elif data['request_type'] == "regenerate-bundle":
+            return RegenerateBundleModel._from_dict(data)
+        raise KeyError("Unsupported request type: %s" % data['request_type'])
+
+    @classmethod
+    def _from_dict(cls, data):
+        args = []
+        kwargs = {}
+        for attr in cls._data_attrs:
+            args.append(data[attr])
+        for kwattr in cls._kwattrs:
+            kwargs[kwattr] = data[kwattr]
+        return cls(*args, **kwargs)
 
     def to_dict(self):
-        return {
-            "id": self.id,
-            "state": self.state,
-            "state_reason": self.reason,
-            "state_history": self.state_history,
-            "from_index": self.from_index,
-            "from_index_resolved": self.from_index,
-            "bundles": self.bundles,
-            "removed_operators": self.removed_operators,
-            "organization": self.organization,
-            "binary_image": self.binary_image,
-            "binary_image_resolved": self.binary_image_resolved,
-            "index_image": self.index_image,
-            "request_type": self.request_type,
-            "arches": self.arches,
-            "bundle_mapping": self.bundle_mapping,
-            "omps_operator_version": self.omps_operator_version,
-        }
+        result = {}
+        for key in self._data_attrs:
+            result[key] = self.data[key]
+        for key in self._kwattrs:
+            result[key] = self.data[key]
+        return result
 
     def __eq__(self, other):
-        if (
-            self.id == other.id
-            and self.state == other.state
-            and self.reason == other.reason
-            and self.state_history == other.state_history
-            and self.from_index == other.from_index
-            and self.from_index_resolved == other.from_index_resolved
-            and self.bundles == other.bundles
-            and self.removed_operators == other.removed_operators
-            and self.organization == other.organization
-            and self.binary_image == other.binary_image
-            and self.binary_image_resolved == other.binary_image_resolved
-            and self.index_image == other.index_image
-            and self.request_type == other.request_type
-            and self.arches == other.arches
-            and self.bundle_mapping == other.bundle_mapping
-            and self.omps_operator_version == other.omps_operator_version
-        ):
-            return True
-        return False
+        return isinstance(other, IIBBuildDetailsModel) and self.data == other.data
```

Comment:
As suggested @midnightercz: This equivalence seems to be tricky. When I use the equivalence for 2 object (function from_dict() ) it fails at comparing data dictionaries. Then it fails again when 2 dictionaries are compared (function to_dict()). 
@lipoja @midnightercz @querti do you have any ideas how to avoid this problem

Reply:
thanks @lipoja, solved